iceoryx_utils/source/posix_wrapper/unix_domain_socket.cpp

```diff
@@ -114,6 +114,11 @@ UnixDomainSocket& UnixDomainSocket::operator=(UnixDomainSocket&& other) noexcept
 {
     if (this != &other)
     {
+        if (destroy().has_error())
```

Comment:
Writing tests for the move c'tor/move assignment operator is currently not possible due to the missing possibilty to check for an invalidate `expected`.

We need an `is_valid` method:
```
TYPED_TEST(IpcChannel_test, MoveAssignmentLeadsToInvalidation)
{
    auto clientResult =
        TestFixture::IpcChannelType::create(anotherGoodName, IpcChannelMode::BLOCKING, IpcChannelSide::SERVER);
    auto movedValue = std::move(clientResult);

    EXPECT_TRUE(clientResult.is_valid());
    EXPECT_FALSE(movedValue.is_valid());
}
```

Reply:
@elBoberido suggested either `has_value`, `is_moved`, `is_specified`, `is_valid`, `has_content`, `not_invalidated`. I like `is_valid ` the most.

What do you think @elfenpiff @budrus?

Reply:
Yup, I agree, as close to the standard as possible:

Hits on CppRef:
* `has_value` 3
* `is_moved` 0
* `is_specified` 0
* `is_valid` 0
* `has_content` 0
* `not_invalidated` 0
* `has_undefined_state` 0

I'm quite fond of `is_undefined`

We can move this discussion to the follow-up pull request.

Reply:
if the `expected` in in an undefined state after moving, we are violating the C++ standard which explicitly tells that the object must be in an unspecified but defined state ;)

Reply:
Move to #542